### ISSUE TITLE
[FR] Tags

### ISSUE THREAD

**Issue by adnan-usman** [AppFlowy-IO/AppFlowy#5912]
### Description

It will be graet if there is a tag system to add docs in specific tag etc 

### Impact

All type of Users

**Comment by Xazin** [comment]
I wonder how you would use the tags?

I can imagine it would be quite easy to add tags to a Document (view) as a new optional attribute, but I'm not sure what features would have to accommodate this new attribute to make it worthwhile?

**Comment by adnan-usman** [comment]
This is How tags can help us  -

1. **Improved Organization:** Tags allow for flexible categorization beyond
traditional folders.
2. **Enhanced Searchability:** Quickly find documents by filtering or
searching with tags.
3. **User-Friendly:** Tags offer intuitive ways to group related content.
4. **Efficient Management:** Bulk tagging and tag suggestions streamline
document handling.
5. **Better Insights:** Tag analytics provide valuable data on content
trends.
6. **Increased Flexibility:** Tags can complement or replace folders,
offering more ways to organize.

**Comment by Xazin** [comment]
The benefits are too generalized, I was asking for actual features that would follow "tags" so that it would be easier to point out whether it'd be worthwhile to implement.

Eg. Do we need to be able to search for documents by one or more tags? Are tags for a document managed per space (eg. Shared, Private), or one list of available Tags per Workspace, and so forth.

Adding a tag to a document achieves nothing by itself, there most likely needs to be some other features as well, my question was what features do you think would be needed along with adding tags to documents, to make it a worthwhile?

**Comment by zetashift** [comment]
> Adding a tag to a document achieves nothing by itself

I agree with this, however one can already add tags through the calendar:
![image](https://github.com/user-attachments/assets/bdce6904-65f6-40e6-8eda-5c80bd9ca1f2)

So there is one form of it already available, I think it's worth it to give a bit more oomph to it:

The usecase is here is, to have more fine-grained categorization than folders/pages. 
Say I have a big note:
```
# Work Log
...stuff...

[[2024-01-01]] Discussed plan A during #meeting
```

Now I could potentially filter blocks through `meeting` if I want to check some meetings stuff.

There is a way to do it already if your `meeting` tag would be a link to a page instead of a tag. LogSeq does this, it treats pages/docs the same as tags.
I think it's worth discussing if AppFlowy could do the same. You'll treat a tag as a doc, but visually `#tag` still remains with some special coloring.

This way you can even add information to the tag/doc itself!
The search rules would then apply the same rules as a doc, but maybe with a different indicator.

It would make linking data in a database/calendar/kanban with docs, that much easier to find. Because in the end tagging things is a great way to quickly re-discover old content.

I do want to note, these benefits mostly come when working mostly on your personal knowledge base. The advantages are less effective in the context of teams working on docs, because well, they probably don't know all the tags or they just want to write rather than organize.

**Comment by CeJienAJPC** [comment]
> The benefits are too generalized, I was asking for actual features that would follow "tags" so that it would be easier to point out whether it'd be worthwhile to implement.
> 
> Eg. Do we need to be able to search for documents by one or more tags? Are tags for a document managed per space (eg. Shared, Private), or one list of available Tags per Workspace, and so forth.
> 
> Adding a tag to a document achieves nothing by itself, there most likely needs to be some other features as well, my question was what features do you think would be needed along with adding tags to documents, to make it a worthwhile?

**The Need for Tags**
- When there are 100+ documents, it would be easier & faster to search by Tags (unless one remembers the file name).
- Tags can help in building a Graph View (like used in Obsidian, Logseq) if Appflowy plans one in the future (highly recommended).
- Tags can make the search for a file easier across Spaces.
- Tags can be used to look for Documents sharing similar/same Tags.
- Tags when used inside the Document (say a specific para), can be used to reference this specific section inside the Document and all the referenced sections from multiple Documents can be viewed.

Hence, using Tags is highly recommended and organises the entire data over time, as it grows. I hope this answers your question. Regards

**Comment by poelzi** [comment]
This is what logseq is known for and what is definitive missing. Writing '#tag' should automatically generate a tag, pages should have tags.
Tag pages should collect all backlinks and allow customization on the upper part. Without tags, the whole note taking becomes useless, since cross linking is a burden.
I can't understand why this is missing.

**Comment by arigit** [comment]
+1 vote on tag support. It's a basic migration blocker. together with diagram embedding (mermaid, draw.io), journal mode, templates. On the plus side, this is the best MD editor I've seen.

**Comment by LucasXu0** [comment]
> If there was an option to add tags it would be perfect.

Similar request from the Google Play Console.

**Comment by poelzi** [comment]
I don't get it. This is a blocker for what 1/3 of people using other programs, because this is so basic functionality for any such program since the last 10 years ?
